consumers/campaigner/main.go

```diff
@@ -44,150 +47,112 @@ func (h *MessageHandler) HandleMessage(m *nsq.Message) error {
-				if err != nil {
-					logrus.WithError(err).
-						WithField("subscriber", s).
-						Error("Unable to get subscriber metadata.")
-
-					continue
-				}
-
-				if s.Name != "" {
-					m["name"] = s.Name
-				}
-
-				url, err := s.GetUnsubscribeURL(msg.UserUUID)
-				if err != nil {
-					logrus.WithError(err).
-						WithField("subscriber", s).
-						Error("Unable to get unsubscribe url.")
-				} else {
-					m["unsubscribe_url"] = url
-				}
-
-				jsonMeta, err := json.Marshal(m)
-				if err != nil {
-					logrus.WithError(err).
-						WithField("subscriber", s).
-						Error("Unable to marshal metadata to json.")
-
-					continue
-				}
-				d := &ses.BulkEmailDestination{
-					Destination: &ses.Destination{
-						ToAddresses: []*string{aws.String(s.Email)},
-					},
-					ReplacementTemplateData: aws.String(string(jsonMeta)),
-				}
-
-				dest = append(dest, d)
-			}
-
-			defaultData, err := json.Marshal(msg.TemplateData)
-			if err != nil {
-				logrus.WithError(err).Error("Unable to marshal template data as JSON.")
-				continue
-			}
-
-			// prepare message for publishing to the queue
-			input := &ses.SendBulkTemplatedEmailInput{
-				Source:              aws.String(msg.Source),
-				Template:            aws.String(msg.Campaign.Template.Name),
-				Destinations:        dest,
-				DefaultTemplateData: aws.String(string(defaultData)),
-				DefaultTags: []*ses.MessageTag{
-					&ses.MessageTag{
-						Name:  aws.String("campaign_id"),
-						Value: aws.String(strconv.FormatInt(msg.Campaign.ID, 10)),
-					},
-					&ses.MessageTag{
-						Name:  aws.String("user_id"),
-						Value: aws.String(msg.UserUUID),
-					},
-				},
-			}
-
-			if msg.ConfigurationSetExists {
-				input.ConfigurationSetName = aws.String(emails.ConfigurationSetName)
-			}
-
-			uuid := uuid.New()
-
-			bulkMsg := entities.BulkSendMessage{
-				UUID:       uuid.String(),
-				Input:      input,
-				CampaignID: msg.Campaign.ID,
-				UserID:     msg.UserID,
-				SesKeys:    &msg.SesKeys,
-			}
-			msg, err := json.Marshal(bulkMsg)
-
-			if err != nil {
-				logrus.WithError(err).Error("Unable to marshal bulk message input.")
-				continue
-			}
-
-			// publish the message to the queue
-			err = h.p.Publish(entities.SendBulkTopic, msg)
-			if err != nil {
-				logrus.WithError(err).Error("Unable to publish message to send bulk topic.")
-			}
+		senderBytes, err := json.Marshal(sender)
+		if err != nil {
+			logrus.WithError(err).Error("Unable to marshal bulk message input.")
+			continue
 		}
 
-		lastSub := subs[len(subs)-1]
-		nextID = lastSub.ID
-		timestamp = lastSub.CreatedAt
+		// publish the message to the queue
+		err = h.p.Publish(entities.SenderTopic, senderBytes)
+		if err != nil {
+			logrus.WithError(err).Error("Unable to publish message to send bulk topic.")
+			continue
+		}
 	}
 
-	c := msg.Campaign
-	c.UserID = msg.UserID
-	c.Status = entities.StatusSent
-	c.CompletedAt.SetValid(time.Now().UTC())
+	// set  vars for next batches
```

Comment:
This should be inside the for loop, so GetDistinctSubscribersBySegmentIDs can be called with the `nextID` and `timestamp`.